waspc/src/Wasp/Analyzer/TypeChecker/TypeError.hs

```diff
@@ -117,3 +97,38 @@ getWeakenErrorMessageAndCtx = getTypeCoercionErrorMessageAndCtx $
       [ concatPrefixAndText "Expected type: " (show t),
         concatPrefixAndText "Actual type:   " (show $ exprType texpr)
       ]
+
+getTypeCoercionErrorMessageAndCtx :: (Type -> TypedExpr -> String) -> TypeCoercionError -> (String, Ctx)
+getTypeCoercionErrorMessageAndCtx getUncoercableTypesMsg typeCoercionError = (fullErrorMsg, ctx)
+  where
+    (typesErrorMsg, ctxMsgs, ctx) = getUncoercableTypesMsgAndCtxMsgsAndCtx getUncoercableTypesMsg typeCoercionError
+    fullErrorMsg
+      | null ctxMsgs = typesErrorMsg
+      | otherwise = intercalate "\n\n" [typesErrorMsg, concatContextualMsgs ctxMsgs]
+
+-- | Takes a function for constructing a type coercion error message, and a `TypeCoercionError`.
```

Comment:
I would first say what it does.

So:
```
For a given a type coercion error, constructs error message and its context.
```

Then you can continue similar to what you have, describing details of how that is done -> we are given a function that constructs the main error message, and what is exactly returned as context is a **list** of messages that describe the structure in which error occurred, ... -> you already have this part will written, I just wanted to suggest adding this "high level overview" sentence as the start of the comment.

Reply:
Definitely, I'll switch it up.